src/main/scala/io/iohk/ethereum/network/p2p/MessageDecoders.scala

```diff
@@ -38,45 +35,87 @@ object NetworkMessageDecoder extends MessageDecoder {
 // scalastyle:off
 object EthereumMessageDecoder extends MessageDecoder {
 
-  override def fromBytes(msgCode: Int, payload: Array[Byte], protocolVersion: Version): Message =
+  private type MessageHandler = PartialFunction[(Int, Array[Byte]), Message]
+
+  override def fromBytes(msgCode: Int, payload: Array[Byte], protocolVersion: Version): Message = {
     (protocolVersion, msgCode) match {
       //wire protocol
       case (_, Hello.code) => payload.toHello
+      case (PV61, _) => pv61Handler(msgCode, payload)
+      case (PV62, _) => pv62Handler(msgCode, payload)
+      case (PV63, _) => pv63Handler(msgCode, payload)
+      case (PV64, _) => pv64Handler(msgCode, payload)
```

Comment:
Won't this mean that if a peer (from mainnet or mordor) supports ETH/ETC's PV64 we'll fail to decode their NewBlock and Status messages?

Wouldn't it be simpler if we rename our PV64 to PV66? Using PV64 might lead to other confusions like this

Reply:
yes it is, but having the assumption that PV_X includes PV_X-1, we cannot use PV66 too